executor-shared/src/main/java/data/Params.java

```
@@ -26,4 +29,14 @@ public String getRequiredParam(String param) {
     public String getOptionalParam(String param, String defaultValue) {
         return params.getOrDefault(param, defaultValue);
     }
+
+    public Date getOptionalParamAsDate(String param, Date defaultValue, SimpleDateFormat dateFormatter) {
+        if (params.containsKey(param)) {
+            try {
+                return dateFormatter.parse(params.get(param));
+            } catch (ParseException ignored) {
```

hmmm..

---

I think we should provide similar methods for all types we use not to repeat casting code

---

I ment empty catch block

---

If we fail to parse we use the default value that is returned at the end of the method.
It's supposed to work like that

---

ok then!